scripts/check-sidebar-lazy-layout.py

```
@@ -1,741 +1,420 @@
+            if marker not in clean:
+                violations.append(f"{filename} is missing required marker {marker}")
+
+    for filename, source in sources_by_name.items():
+        clean = neutralize_swift(source)
+        for callback in LAYOUT_CALLBACKS:
+            for body in extract_all_function_bodies(clean, callback):
+                for pattern in LAYOUT_MUTATION_PATTERNS:
+                    if pattern.search(body):
+                        violations.append(
+                            f"{filename}.{callback} mutates/reconfigures the table from a layout callback"
+                        )
```

Ruff (0.15.21)

[warning] Use `list.extend` to create a transformed list

(PERF401)